### PR TITLE
Revamp README.md

### PR DESCRIPTION
**Pull request by FallenChromium**
* [x] Read PR [documentation](https://github.com/ostis-ai/sc-machine/blob/main/docs/dev/pr.md/)
* [ ] Update changelog  (doesn't seem like a part of the project itself, so I decided not to include it in the changelog)
* [x] Update documentation

Planned: additional gif files or illustrations demonstrating:
- the KG-building and agent-based processing mechanisms (will replace the "sc-machine contains" section)
- Full usage example (with data operations visualised separately) - the demo section will be added: 
  1. writing a new source file
  2. launching the system
  3. calling a processing agent 
  4. getting a response

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,79 +1,97 @@
...
+  - core concepts
+  - rationale behind sc-machine
+  - system design
+  - software interfaces
+  
+  is redistributed in a form of [SCn-TeX document](https://github.com/ostis-ai/ostis-web-platform/blob/develop/docs/main.pdf).
+
+## Quick start
+
+sc-machine is a core of any ostis-system, so you can use a reference sytem named [OSTIS Web platform](https://github.com/ostis-ai/ostis-web-platform) to get sc-machine up and running quickly.
```
Comment (MikhailSadovsky) [inline README.md:51]:
```suggestion
sc-machine is a core of any ostis-system, so you can use a reference system named [OSTIS Web platform](https://github.com/ostis-ai/ostis-web-platform) to get sc-machine up and running quickly.
```

Comment (FallenChromium) [reply README.md]: Oops. Thanks! I've fixed it.

**Review by VasKho — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -1,79 +1,97 @@
...
+(programs, modules, systems, communication between systems) knowledge is represented using the same language: SC-code.
+
+</details>
+
+## Documentation
+
+- A brief user manual and developer docs are hosted on our [GitHub Pages](https://ostis-ai.github.io/sc-machine).
+- Full documentation, including:
+  - core concepts
+  - rationale behind sc-machine
```
Comment (NikitaZotov) [inline README.md:42]:
```suggestion
  - components implementation
  - sc-model architecture and sc-memory model
```

File: README.md [hunk 3]
```diff
@@ -1,79 +1,97 @@
...
+  - system design
```
Comment (NikitaZotov) [inline README.md:43]:
```suggestion
  - ontology-based design of sc-machine
```

File: README.md [hunk 4]
```diff
@@ -1,79 +1,97 @@
...
+  - software interfaces
+  
+  is redistributed in a form of [SCn-TeX document](https://github.com/ostis-ai/ostis-web-platform/blob/develop/docs/main.pdf).
+
+## Quick start
+
+sc-machine is a core of any ostis-system, so you can use a reference system named [OSTIS Web platform](https://github.com/ostis-ai/ostis-web-platform) to get sc-machine up and running quickly.
```
Comment (NikitaZotov) [inline README.md:50]:
```suggestion
Sc-machine is a core of any ostis-system, so you can use a reference system named [OSTIS Web platform](https://github.com/ostis-ai/ostis-web-platform) to get sc-machine up and running quickly.
```

File: README.md [hunk 5]
```diff
@@ -1,79 +1,97 @@
...
-```sh
-python3 scripts/healthcheck.py --host <sc-server host> --port <sc-server port>
-``` 
-
-To get more information about these scripts use `-h` flags upon their launch.
+Short version: **sc-machine** is an agent-based knowledge graph processing machine.
+
+The knowledge graphs are represented in SC-code (language of universal knowledge representation) provided by the [**OSTIS Technology**](https://github.com/ostis-ai/ostis-project).
+
+sc-machine allows you to integrate problem solutions from different subject areas **by using the same**:
```
Comment (NikitaZotov) [inline README.md:12]:
```suggestion
Sc-machine allows you to integrate problem solutions from different subject areas **by using the same**:
```

File: README.md [hunk 6]
```diff
@@ -1,79 +1,97 @@
...
-./bin/sc-server -c config/sc-machine.ini
-``` 
-
-You can use `healthcheck.py` script to check memory state in sc-server.
-```sh
-python3 scripts/healthcheck.py --host <sc-server host> --port <sc-server port>
-``` 
-
-To get more information about these scripts use `-h` flags upon their launch.
+Short version: **sc-machine** is an agent-based knowledge graph processing machine.
```
Comment (NikitaZotov) [inline README.md:8]:
```suggestion
Short version: **Sc-machine** is an agent-based knowledge graph processing machine.
```

Comment (NikitaZotov) [inline README.md:8]:
```suggestion
Short version: **sc-machine** is an agent-based knowledge graph processing machine. It is a base program variant of computer system interpreter which fully emulates semantic computer!
```

File: README.md [hunk 7]
```diff
@@ -1,79 +1,97 @@
...
+
+The knowledge graphs are represented in SC-code (language of universal knowledge representation) provided by the [**OSTIS Technology**](https://github.com/ostis-ai/ostis-project).
```
Comment (NikitaZotov) [inline README.md:10]:
```suggestion
Sc-machine stores and process the knowledge graphs represented in SC-code (language of universal knowledge representation) and is used to develop, design and exercise new generation computer systems. All ontology-based design of sc-machine is provided by the [**OSTIS Technology**](https://github.com/ostis-ai/ostis-project).
```

File: README.md [hunk 8]
```diff
@@ -1,79 +1,97 @@
...
+- `sc-event-manager` event-based processing (async program workflow handling)
+- tools to communicate with `sc-memory`:
+  - native C++ API
+  - `sc-server` - a Websocket server that provides a network API to communicate with storage using JSON
+  - `sc-builder` loads information from SCs files into storage (SCs-code is a one of external representation languages of SC-code)
+
+<details>
+   <summary>More info</summary>
+
+sc-machine is a **platform-independent graph database management system** that can store / retrieve knowledge graphs and run tasks (agents) on them.
```
Comment (NikitaZotov) [inline README.md:30]:
```suggestion
Sc-machine is a **platform-independent graph database management system** that can store / retrieve knowledge graphs and run tasks (agents) on them.
```

File: README.md [hunk 9]
```diff
@@ -1,79 +1,97 @@
...
+sc-machine allows you to integrate problem solutions from different subject areas **by using the same**:
+
+- **technology**
+- **programming and data representation language**
+- **software stack**.
+
+This project contains:
+
+- `sc-memory` - semantic storage to store graph constructions
+- `sc-event-manager` event-based processing (async program workflow handling)
```
Comment (NikitaZotov) [inline README.md:21]:
```suggestion
- `sc-event-manager` - event-based processing (async program workflow handling)
```

File: README.md [hunk 10]
```diff
@@ -1,79 +1,97 @@
...
+
+sc-machine allows you to integrate problem solutions from different subject areas **by using the same**:
+
+- **technology**
+- **programming and data representation language**
+- **software stack**.
+
+This project contains:
+
+- `sc-memory` - semantic storage to store graph constructions
```
Comment (NikitaZotov) [inline README.md:20]:
```suggestion
- `sc-memory` - semantic storage to store graph constructions;
```

File: README.md (hunk 9 above)
Comment (NikitaZotov) [inline README.md:21]:
```suggestion
- `sc-event-manager` event-based processing (async program workflow handling);
```

File: README.md [hunk 11]
```diff
@@ -1,79 +1,97 @@
...
+- `sc-event-manager` event-based processing (async program workflow handling)
+- tools to communicate with `sc-memory`:
+  - native C++ API
+  - `sc-server` - a Websocket server that provides a network API to communicate with storage using JSON
+  - `sc-builder` loads information from SCs files into storage (SCs-code is a one of external representation languages of SC-code)
```
Comment (NikitaZotov) [inline README.md:25]:
```suggestion
  - native C++ API;
  - `sc-server` - a Websocket server that provides a network API to communicate with storage using JSON;
  - `sc-builder` loads information from SCs files into storage (SCs-code is a one of external representation languages of SC-code).
```

File: README.md [hunk 12]
```diff
@@ -1,79 +1,97 @@
...
-``` 
-
-To get more information about these scripts use `-h` flags upon their launch.
+Short version: **sc-machine** is an agent-based knowledge graph processing machine.
+
+The knowledge graphs are represented in SC-code (language of universal knowledge representation) provided by the [**OSTIS Technology**](https://github.com/ostis-ai/ostis-project).
+
+sc-machine allows you to integrate problem solutions from different subject areas **by using the same**:
+
+- **technology**
```
Comment (NikitaZotov) [inline README.md:14]:
```suggestion
- **technology**,
```

**Review by NikitaZotov — CHANGES_REQUESTED** [review]

File: README.md (hunk 6 above)
Comment (FallenChromium) [reply README.md:8]: It looks ugly :(

File: README.md (hunk 2 above)
Comment (FallenChromium) [reply README.md:42]: I haven't heard about `sc-model` earlier. 

Anyway, this suggestion and the one below can be generalised to a single one - replacing more abstract terms understood by the industry with ones specific to our project. But I do not really agree with that. The complexity of the terms used in the README was pretty much the reason why I've made the change - this document was too hard to be understood by a person which has never interacted with OSTIS before.
Basically, `system design` is `SC model architecture + sc-memory model + implementation details`, so why wouldn't we want to use a more eloquent term to express it?

File: README.md (hunk 6 above)
Comment (FallenChromium) [reply README.md:8]:
```suggestion
Short version: **sc-machine** is a software package aimed to emulate semantic computer behaviour. It uses agent-based approach to to process knowledge graphs.
```
That would both be more informative and clear. Though we should probably provide a link to a "semantic computer", for example [this one](https://en.wikipedia.org/wiki/Semantic_computing). What do you think?

File: README.md (hunk 7 above)
Comment (FallenChromium) [reply README.md:10]: `ontology-based design of sc-machine` - what do you mean by that?
`exercise new generation computer systems` - ...and that.
I agree that a bit more info should be added, but I am not sure what did you have in mind with these two. My current version would be something along those lines:
```suggestion
sc-machine stores and processes knowledge graphs represented in SC-code (language of universal knowledge representation) provided by the [**OSTIS Technology**](https://github.com/ostis-ai/ostis-project).
```

Comment (Liza909) [inline README.md:10]:
```suggestion
The knowledge graphs are represented in the SC-code (language of the universal knowledge representation) provided by the [**OSTIS Technology**](https://github.com/ostis-ai/ostis-project).
```

File: README.md (hunk 5 above)
Comment (Liza909) [inline README.md:12]:
```suggestion
sc-machine allows integrating problem solutions from different subject domains **by using the same**:
```

File: README.md [hunk 13]
```diff
@@ -1,79 +1,97 @@
...
+- **software stack**.
+
+This project contains:
+
+- `sc-memory` - semantic storage to store graph constructions
+- `sc-event-manager` - event-based processing (async program workflow handling)
+- tools to communicate with `sc-memory`:
+  - native C++ API
+  - `sc-server` - a Websocket server that provides a network API to communicate with storage using JSON
+  - `sc-builder` loads information from SCs files into storage (SCs-code is a one of external representation languages of SC-code)
```
Comment (Liza909) [inline README.md:25]:
```suggestion
  - `sc-builder` loads information from SCs files into the storage (SCs-code is one of the external representation languages of the SC-code)
```

File: README.md [hunk 14]
```diff
@@ -1,79 +1,97 @@
...
+
+<details>
+   <summary>More info</summary>
+
+sc-machine is a **platform-independent graph database management system** that can store / retrieve knowledge graphs and run tasks (agents) on them.
+
+Both declarative (data, data structures, documentation, tasks specification, etc.) and procedural
+(programs, modules, systems, communication between systems) knowledge is represented using the same language: SC-code.
```
Comment (Liza909) [inline README.md:33]:
```suggestion
(programs, modules, systems, communication between systems) knowledge is represented using the same language: the SC-code.
```

File: README.md [hunk 15]
```diff
@@ -1,79 +1,97 @@
...
+
+</details>
+
+## Documentation
+
+- A brief user manual and developer docs are hosted on our [GitHub Pages](https://ostis-ai.github.io/sc-machine).
+- Full documentation, including:
+  - core concepts
+  - rationale behind sc-machine
```
Comment (Liza909) [inline README.md:42]:
```suggestion
  - rationale behind the sc-machine
```

File: README.md [hunk 16]
```diff
@@ -1,79 +1,97 @@
...
+  - system design
+  - software interfaces
+  
+  is redistributed in a form of [SCn-TeX document](https://github.com/ostis-ai/ostis-web-platform/blob/develop/docs/main.pdf).
```
Comment (Liza909) [inline README.md:46]:
```suggestion
  is redistributed in a form of the [SCn-TeX document](https://github.com/ostis-ai/ostis-web-platform/blob/develop/docs/main.pdf).
```

File: README.md [hunk 17]
```diff
@@ -1,79 +1,97 @@
...
+
+## Quick start
+
+sc-machine is a core of any ostis-system, so you can use a reference system named [OSTIS Web platform](https://github.com/ostis-ai/ostis-web-platform) to get sc-machine up and running quickly.
```
Comment (Liza909) [inline README.md:50]:
```suggestion
sc-machine is a core of any ostis-system, so you can use a reference system named [OSTIS Web platform](https://github.com/ostis-ai/ostis-web-platform) to get the sc-machine up and running quickly.
```

File: README.md [hunk 18]
```diff
@@ -1,79 +1,97 @@
...
+
+## Installation
+
+- Docker:
+  We're providing a Docker image for this project. Head to [Installing with Docker](https://ostis-ai.github.io/sc-machine/docker) to learn more. It's the recommended way to deploy sc-machine.
```
Comment (Liza909) [inline README.md:55]:
```suggestion
  We provide a Docker image for this project. Head to [Installing with Docker](https://ostis-ai.github.io/sc-machine/docker) to learn more. It's the recommended way to deploy the sc-machine.
```

File: README.md [hunk 19]
```diff
@@ -1,79 +1,97 @@
...
+- Native:
+  If you do not have the option to deploy the system using Docker please refer to the docs for your OS:
+  [Build on Linux](https://ostis-ai.github.io/sc-machine/build/linux-build/) or [Build on macOS](https://ostis-ai.github.io/sc-machine/build/osx-build/) respectively.
+
+  Note: currently, sc-machine isn't *natively* supported on Windows.
```
Comment (Liza909) [inline README.md:60]:
```suggestion
  Note: currently, the sc-machine isn't *natively* supported on Windows.
```

File: README.md [hunk 20]
```diff
@@ -1,79 +1,97 @@
...
+  ./bin/sc-server -c config/sc-machine.ini
+  ```
+
+Most of these commands have a help page bundled inside, so if you have any questions or want to customize the command behavior, use `--help` flag to learn more about them.
 
 ## Config
 
-This repository provides a default configuration for sc-machine. To customize *sc-machine* usage you can 
-[create your own config file](https://ostis-ai.github.io/sc-machine/other/config).
+This repository provides a default configuration for sc-machine. To customize *sc-machine* to suit your needs you can [create your own config file](https://ostis-ai.github.io/sc-machine/other/config).
```
Comment (Liza909) [inline README.md:86]:
```suggestion
This repository provides a default configuration for the sc-machine. To customize the *sc-machine* to suit your needs you can [create your own config file](https://ostis-ai.github.io/sc-machine/other/config).
```

File: README.md [hunk 21]
```diff
@@ -1,79 +1,97 @@
...
+
+## Feedback
+
+Contributions, bug reports and feature requests are welcome!
```
Comment (Liza909) [inline README.md:90]:
```suggestion
Contributions, bug reports, and feature requests are welcome!
```

File: README.md [hunk 22]
```diff
@@ -1,79 +1,97 @@
...
+## Quick start
+
+sc-machine is a core of any ostis-system, so you can use a reference system named [OSTIS Web platform](https://github.com/ostis-ai/ostis-web-platform) to get sc-machine up and running quickly.
+
+## Installation
+
+- Docker:
+  We're providing a Docker image for this project. Head to [Installing with Docker](https://ostis-ai.github.io/sc-machine/docker) to learn more. It's the recommended way to deploy sc-machine.
+- Native:
+  If you do not have the option to deploy the system using Docker please refer to the docs for your OS:
```
Comment (Liza909) [inline README.md:57]:
```suggestion
  If you do not have the option to deploy the system using Docker, please refer to the docs for your OS:
```

**Review by KovalM — CHANGES_REQUESTED** [review]

File: README.md (hunk 6 above)
Comment (NikitaZotov) [reply README.md:8]: You can find information about semantic computer in the OSTIS Standart. And I don't think that sc-machine is a package, the better to write that sc-machine is a sc-core of base program variant of computer system interpreter which fully emulates semantic computer!

File: README.md (hunk 7 above)
Comment (NikitaZotov) [reply README.md:10]: We can discuss this and I told you about main ideas of sc-machine in key in which linked concepts are formalized in the OSTIS Standard

File: README.md (hunk 8 above)
Comment (NikitaZotov) [reply README.md:30]: I propose discuss it :)

File: README.md (hunk 12 above)
Comment (NikitaZotov) [reply README.md:14]: Lets leave without comma. You are right.

File: README.md (hunk 2 above)
Comment (NikitaZotov) [reply README.md:42]: I would like to express myself in terms and concepts interpreted within the framework of OSTIS technology. Documentation must be represented in these terms and concepts

Comment (KovalM) [reply README.md:42]: After discuss: leave Pavel's version before approving core concepts of SC-machine

File: README.md (hunk 5 above)
Comment (KovalM) [reply README.md:12]:
```suggestion
SC-machine allows you to integrate problem solutions from different subject areas **by using the same**:
```

File: README.md (hunk 6 above)
Comment (KovalM) [reply README.md:8]: After discuss: apply Pave's variant from this conversation

File: README.md (hunk 7 above)
Comment (KovalM) [reply README.md:10]: After discuss
```suggestion
SC-machine stores and processes knowledge graphs represented in SC-code (language of universal knowledge representation). All ontology-based design of sc-machine is provided by the [**OSTIS Technology**](https://github.com/ostis-ai/ostis-project).
```

Comment (KovalM) [reply README.md:10]: After discuss: apply Pavel's variant from this conversation

File: README.md (hunk 21 above)
Comment (FallenChromium) [reply README.md:90]: Oh, Oxford comma? Classy :)
I assume our students may misinterpret it as a mistake (even though it's a stylistic/optional punctuation), so I'd choose to omit it in this case 🤔

Comment (Liza909) [reply README.md:90]: There are Oxford commas in one of your PR's, so I just propose to unificate such syntactic constructions)

Comment (FallenChromium) [reply README.md:90]: It looks like I've missed it then, it wasn't my intention. Can you send a link please?

**Review by NikitaZotov — APPROVED** [review]

**Review by NikitaZotov — APPROVED** [review]
